admin/links/class-link-reindex-dashboard.php

```diff
@@ -0,0 +1,183 @@
+				);
+			}
+			else {
+				$progress = sprintf(
+				/* translators: 1: expands to a <span> containing the number of items recalculated. 2: expands to a <strong> containing the total number of items. */
+					__( 'Post %1$s of %2$s analyzed.', 'wordpress-seo' ),
+					sprintf( '<span id="wpseo_count_index_links_%s">0</span>', esc_attr( $post_type ) ),
+					sprintf( '<strong id="wpseo_count_%s_total">%d</strong>', esc_attr( $post_type ), $this->unprocessed[ $post_type ] )
+				);
+
+				$inner_text = sprintf( '<div id="wpseo_index_links_%s_progressbar" class="wpseo-progressbar"></div>', esc_attr( $post_type ) );
+				$inner_text .= sprintf( '<p>%s</p>', $progress );
+			}
+
+			$blocks[] = sprintf( '<div><p>%s</p>%s</div>',
+				/* Translators: %s resolves to the post type label. */
+				esc_html( sprintf( __( 'Detecting links for %s...', 'wordpress-seo' ), $post_type_label ) ),
+				$inner_text
+			);
+		}
+
+		?>
+		<div id="wpseo_index_links_wrapper" class="hidden">
+			<?php echo implode( '<hr />', $blocks ); ?>
+			<button onclick="tb_remove();" type="button"
+					class="button"><?php _e( 'Stop analyzing', 'wordpress-seo' ); ?></button>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Enqueues site wide analysis script
+	 */
+	public function enqueue() {
+		$asset_manager = new WPSEO_Admin_Asset_Manager();
+		$asset_manager->enqueue_script( 'reindex-links' );
+
+		$data = array(
+			'amount'  => $this->unprocessed,
+			'restApi' => array(
+				'root'     => esc_url_raw( rest_url() ),
+				'endpoint' => WPSEO_Link_Reindex_Post_Endpoint::REST_NAMESPACE . '/' . WPSEO_Link_Reindex_Post_Endpoint::ENDPOINT_QUERY,
+				'nonce'    => wp_create_nonce( 'wp_rest' ),
+			),
+			'message' => array(
+				'indexingCompleted' => $this->message_already_indexed(),
+			),
+			'l10n'    => array(
+				'calculationInProgress' => __( 'Calculation in progress...', 'wordpress-seo' ),
+				'calculationCompleted'  => __( 'Calculation completed.', 'wordpress-seo' ),
+			),
+		);
+
+		wp_localize_script( WPSEO_Admin_Asset_Manager::PREFIX . 'reindex-links', 'yoastReindexLinksData', array( 'data' => $data ) );
+	}
+
+
+	/**
+	 * Checks if the current page is the dashboard page.
+	 *
+	 * @return bool True when current page is the dashboard page.
+	 */
+	protected function is_dashboard_page() {
+		return ( filter_input( INPUT_GET, 'page' ) === 'wpseo_dashboard' );
+	}
+
+	/**
+	 * When everything has been indexed already.
+	 *
+	 * @return string
+	 */
+	protected function message_already_indexed() {
+		return '<span class="wpseo-checkmark-ok-icon"></span>' . esc_html__( 'Good job! You\'ve optimized your internal linking suggestions.', 'wordpress-seo' );
+	}
+
+	/**
+	 * Filters the post types to remove unwanted items.
+	 *
+	 * @param string $public_post_type The post type to filter.
+	 *
+	 * @return bool Returns true if it is kept, false if removed.
+	 */
+	protected function filter_post_types( $public_post_type ) {
```

Comment:
I expect a function called filter_post_types to actually filter the posts, not only to return whether the passed type is unwanted or not.